scipy/optimize/_numdiff.py

```diff
@@ -570,3 +638,48 @@ def check_derivative(fun, jac, x0, bounds=(-np.inf, np.inf), args=(),
                                    args=args, kwargs=kwargs)
         abs_err = np.abs(J_to_test - J_diff)
         return np.max(abs_err / np.maximum(1, np.abs(J_diff)))
+
+
+class FiniteDifference:
+    """Represents finite difference approximation.
```

Comment:
What about "Finite-difference approximation of first-order derivatives."

I think it is better because the if the function ``fun``returns a scalar it would be the gradient and not the Jacobian matrix that would be approximated. Or do you think I am overcomplicating?

About the class name: I totally agree with you, ``FiniteDifference`` is not a very good name, but I didn't manage to come up with nothing better so far.